### ISSUE TITLE
[ TESTS ] for general functions and views

### ISSUE THREAD

**Issue by vsoch** [singularityhub/sregistry#8]
This is harder to do with a web based application, but we will need to have tests for confidence that the application continues to work.

**Comment by remyd1** [comment]
Hi @vsoch 

I just saw this feature request. This is usually done with CI tests and a build farm (like github with travis-ci).

IMHO, you should do it as a plugin first, like you did for security scans with clair.

You can look at buildbot, gitlab-ci and jenkins: https://www.digitalocean.com/community/tutorials/ci-cd-tools-comparison-jenkins-gitlab-ci-buildbot-drone-and-concourse

BTW you can also just check the `%test` section.

Best regards,
Rémy

**Comment by vsoch** [comment]
Yes this would be great! I put the help-wanted because one dinosaur can only take on so much... #tiny-arms

**Comment by vsoch** [comment]
I've done CI with Travis, Circle (and a bit of Jenkins) numerous times, but just haven't gotten around to something similar for sregistry. Minimally we would want something similar (and free) and then an additional option for a plugin (for users to run tests on their infrastructure).
